### PR TITLE
Mehmet B. React HW Week2

### PR DESCRIPTION
**Pull request by mehmet-bozdemir**

### CODE REVIEW

File: homework-submission/react-homework/src/Components/DynamicList.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+class TodoItem extends Component {
+  render() {
+    return (
+      <div className="todoItem">
+        <span
+          onClick={this.props.click}
+          className={`todoItem-name ${this.props.done ? ' line-through' : ''}`}
+        >
+          {this.props.description}, {this.props.deadline} {this.props.done}
```
Comment (ArcoMul) [inline homework-submission/react-homework/src/Components/DynamicList.js]: Destructering these values from `this.props` would make your code more readable

File: homework-submission/react-homework/src/Components/DynamicList.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    );
+  }
+}
+
+class Form extends Component {
+  render() {
+    return (
+      <div>
+        <form onSubmit={this.props.click}>
+          <input type="number" name="id" placeholder="ïd" className="item" />
```
Comment (ArcoMul) [inline homework-submission/react-homework/src/Components/DynamicList.js]: Having to enter an id yourself when adding a todo is a bit strange. You can generate an id using code, give it a try.

File: homework-submission/react-homework/src/Components/DynamicList.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+        {
+          id: 3,
+          description: 'Eat breakfast',
+          deadline: '2017-09-09',
+          done: false
+        }
+      ]
+    };
+    this.deleteItemHandler = this.deleteItemHandler.bind(this);
+    this.onSubmitHandle = this.onSubmitHandle.bind(this);
```
Comment (ArcoMul) [inline homework-submission/react-homework/src/Components/DynamicList.js]: These methods are arrow functions, no need to bind `this`

**Comment by mehmet-bozdemir** [comment]
Dear Arco, thanks for your feedback, I am gonna implement them today.

**Comment by mehmet-bozdemir** [comment]
Dear Arco, I implemented feedback from week2 and carried new version to
week 3, I don't know if the implementation of week 2 is ok.
Regards.

**Comment by ArcoMul** [comment]
Nice improvements @mehmet-bozdemir :-) And you code looks good :+1: It's still a bit strange that you have to fill in the value if a todo is done or not (instead of a checkbox), but that wasn't specifically part oft the homework.

**Pull request closed without merging**